database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractAnnotationFromVariant.java

```diff
@@ -0,0 +1,346 @@
+        }
+    }
+
+    private UpdateOneModel<Document> buildUpdateDocument(Document variantDocument, Document annotationSubdocument) {
+        Set<Integer> soSet = computeSoSet(annotationSubdocument, CONSEQUENCE_TYPE_FIELD, SO_FIELD);
+        Set<String> xrefSet = computeXrefSet(annotationSubdocument, XREFS_FIELD, XREF_ID_FIELD);
+        List<Double> sift = computeMinAndMaxScore(annotationSubdocument, SIFT_FIELD);
+        List<Double> polyphen = computeMinAndMaxScore(annotationSubdocument, POLYPHEN_FIELD);
+
+        Document newAnnotationSubdocument = new Document()
+                .append(VEP_VERSION_FIELD, databaseParameters.getVepVersion())
+                .append(CACHE_VERSION_FIELD, databaseParameters.getVepCacheVersion());
+
+        if (!soSet.isEmpty()) {
+            newAnnotationSubdocument.append(SO_FIELD, soSet);
+        }
+        if (!xrefSet.isEmpty()) {
+            newAnnotationSubdocument.append(XREFS_FIELD, xrefSet);
+        }
+        if (!sift.isEmpty()) {
+            newAnnotationSubdocument.append(SIFT_FIELD, sift);
+        }
+        if (!polyphen.isEmpty()) {
+            newAnnotationSubdocument.append(POLYPHEN_FIELD, polyphen);
+        }
+
+        List<Document> newAnnotationArray = Collections.singletonList(newAnnotationSubdocument);
+
+        Document query = new Document(ID_FIELD, variantDocument.get(ID_FIELD));
+        Bson update = Updates.set(ANNOT_FIELD, newAnnotationArray);
+        return new UpdateOneModel<>(query, update);
+    }
+
+    private Set<Integer> computeSoSet(Document originalAnnotationField, String outerField, String innerField) {
+        Set<Integer> soSet = new TreeSet<>();
+
+        List<Document> cts = (List<Document>) originalAnnotationField.get(outerField);
+        if (cts != null) {
+            for (Document ct : cts) {
+                Object sos = ct.get(innerField);
+                if (sos != null) {
+                    soSet.addAll((List<Integer>) sos);
+                }
+            }
+        }
+
+        return soSet;
+    }
+
+    private Set<String> computeXrefSet(Document originalAnnotationField, String outerField, String innerField) {
+        Set<String> xrefSet = new TreeSet<>();
+
+        List<Document> cts = (List<Document>) originalAnnotationField.get(outerField);
+        if (cts != null) {
+            for (Document ct : cts) {
+                String xref = ct.getString(innerField);
+                if (xref != null) {
+                    xrefSet.add(xref);
+                }
+            }
+        }
+
+        return xrefSet;
+    }
+
+    private List<Double> computeMinAndMaxScore(Document originalAnnotationField, String scoreType) {
+        Double min = Double.POSITIVE_INFINITY;
+        Double max = Double.NEGATIVE_INFINITY;
+        boolean thereIsAtLeastOneScore = false;
+
+        List<Document> cts = (List<Document>) originalAnnotationField.get(CONSEQUENCE_TYPE_FIELD);
```

Comment:
This declaration is easier to understand than the equivalents in `computeSoSet` and `computeXrefSet`. Maybe those 2 methods could get also rid of the `outerField` parameter and use the constant instead?